erpnext/stock/doctype/delivery_note/delivery_note.py

```diff
@@ -628,3 +634,26 @@ def make_sales_return(source_name, target_doc=None):
 def update_delivery_note_status(docname, status):
 	dn = frappe.get_doc("Delivery Note", docname)
 	dn.update_status(status)
+
+@frappe.whitelist()
+def email_coas(docname):
+	delivery_note = frappe.get_doc("Delivery Note", docname).as_dict()
+	if not delivery_note.get("contact_email"):
+		frappe.throw(_("No contact email found"))
+
+	coas = [item.get("certificate_of_analysis") for item in delivery_note.get("items") if item.get("certificate_of_analysis")]
+	if not coas:
+		frappe.msgprint(_("No Certificates of Analysis attached"))
+		return
+
+	attachments = []
+	for item in delivery_note.get("items"):
+		coa = frappe.db.get_value("File", {"file_url": item.certificate_of_analysis},
+			["name", "attached_to_doctype", "attached_to_name"], as_dict = 1)
+		attachments.append({
+			"fid": coa.name
+		})
```

Comment:
If you don't need the extra fields, this can just be:

```suggestion
		coa_file_id = frappe.db.get_value("File", {"file_url": item.certificate_of_analysis}, "name")
		attachments.append({"fid": coa_file_id})
```